File: blind75/go/twosum.go
func twoSum(nums []int, target int) []int {
    numMap := make(map[int]int)
    for i, num := range nums {
        complement := target - num
        if index, ok := numMap[complement]; ok {
            return []int{index, i}
        }
        numMap[num] = i
    }
    return nil
}

File: blind75/php/twosum.php
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $numMap = array();
        for ($i = 0; $i < count($nums); $i++) {
            $complement = $target - $nums[$i];
            if (array_key_exists($complement, $numMap)) {
                return array($numMap[$complement], $i);
            }
            $numMap[$nums[$i]] = $i;
            
        }
        
    }
}
